src/utils/appDataClient.ts

```diff
@@ -1,44 +1,129 @@
-import { Hotspot, PocReceiptsV1 } from '@helium/http'
+import Client, {
+  AnyTransaction,
+  Hotspot,
+  Network,
+  PocReceiptsV1,
+  ResourceList,
+} from '@helium/http'
 import { heliumHttpClient } from '@helium/react-native-sdk'
+import Config from 'react-native-config'
 import { fromNow } from './timeUtils'
 import { getAddress } from './secureAccount'
 
-export const submitTxn = async (txn: string) => {
-  return heliumHttpClient.transactions.submit(txn)
-}
+const privateHeliumClient = new Client(
+  new Network({
+    baseURL: Config.HELIUM_SHADOW_URL,
+    version: 1,
+  }),
+)
 
 export const getHotspotDetails = async (address: string): Promise<Hotspot> => {
-  return heliumHttpClient.hotspots.get(address)
+  try {
+    return await privateHeliumClient.hotspots.get(address)
+  } catch (error) {
+    try {
+      return await heliumHttpClient.hotspots.get(address)
+    } catch (err) {
+      throw new Error(
+        `Helium API call is failed while getting the hotspot details: ${err}`,
```

Comment:
Helium API request failed while trying to retrieve hotspots for address {address}